android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
-    final Map<String, Boolean> arguments = call.arguments();
-    Boolean enabled = arguments.get("logLevel");
-
-    if (enabled == null) {
-      enabled = true;
+
+    private void sendInteraction(String interactionPath, HashMap properties, final Result result) {
+        final OneRequest sendInteractionRequest = new OneRequest.Builder()
+                .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
+                .properties(properties)
+                .build();
+        executor.submit(() -> {
+            OneResponse response;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                try {
+                    response = One.sendInteraction(sendInteractionRequest).join();
+                    One.processResponse(response);
+                    HashMap<String, Object> responseMap = responseObjectToHashMap(response);
+                    result.success(responseMap);
+                } catch (ExecutionException error) {
+                    result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] Completion Error: " + error.getCause());
+                } catch (OneSDKError error) {
+                    result.error(Integer.toString(error.getSystemCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] SDK Error: " + error.getErrorMessage());
+                } catch (OneAPIError error) {
+                    result.error(Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] Api Error: " + error.getErrorMessage());
+                }
+            } else {
+                try {
+                    response = One.sendInteractionLegacySupport(sendInteractionRequest).join();
+                    One.processResponse(response);
+                    HashMap<String, Object> responseMap = responseObjectToHashMap(response);
+                    result.success(responseMap);
+                } catch (ExecutionException error) {
+                    result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+                    Log.e(LOG_TAG, "[Thunderhead] Execution Exception Error: " + error.getLocalizedMessage());
+                    error.printStackTrace();
+                }
+            }
+        });
     }
 
-    if (enabled) {
-      oneLoggingConfiguration = OneLoggingConfiguration.builder()
-              .log(LogLevel.VERBOSE)
-              .log(LogLevel.DEBUG)
-              .log(Component.ANY)
-              .build();
-    } else {
-      oneLoggingConfiguration = OneLoggingConfiguration.builder()
-              .log(LogLevel.WARN)
-              .log(LogLevel.ERROR)
-              .log(Component.ANY)
-              .build();
+    private void sendResponseCode(String responseCode, String interactionPath, final Result result) {
```

Comment:
doesn't `responseCode` and `interactionPath` need to be final since they are used in the runnable?